scipy-1.0/minimize.tex

```diff
@@ -0,0 +1,118 @@
+    \hline
+    Local convergence& & & L & S & S & Q & S & Q & ?  \\
+    \hline
+    Global convergence & & &  & \cmark & \cmark & \cmark & \cmark & \cmark & ?  \\
+    \hline
+    \makecell{Line-search (LS) or\\ trust-region (TR)} & Neither  & LS & LS & LS & LS & TR & TR & TR & TR \\
+    \hline
+    References & \cite{nelder_simplex_1965, wright_direct_1996} & \cite{powell_efficient_1964} &
+      \cite{polak_note_1969, nocedal_numerical_2006} & \cite{nocedal_numerical_2006} & \cite{nocedal_numerical_2006}  &
+      \cite{powell_new_1970, nocedal_numerical_2006} &  \cite{steihaug_conjugate_1983, nocedal_numerical_2006} &
+      \cite{conn_trust_2000, more_computing_1983} & \cite{gould_solving_1999, lenders_trlib:_2016} \\
+    \hline
+  \end{tabular}
+  \caption{Optimization methods from \texttt{minimize} for unconstrained minimization.  The field \textit{It is a wrapper?}
+    informs if the implementation available in SciPy is a wrapper to a function written in other programming language
+    (e.g. C or FORTRAN). The fields \textit{\nth{1}} and \textit{\nth{2} derivatives}
+    informs if the first or second order derivatives are required. When \textit{\nth{2} derivatives} is flagged
+    with the word \textit{Approx.} it means that, while the algorithm does not requires the second-order derivatives from
+    the user, it computes an approximation internally and uses it to accelerate the method convergence.
+    \textit{Local convergence} gives
+    a lower bound to the rate of convergence of the iterations sequence once the iterate is close enough to the
+    objective function solution. The convergence rates mentioned at the table
+    are, from the slowest to the fastest:  linear (L), superlinear (S) and quadratic (Q). The \textit{global convergence}
+    is marked for the algorithms for which there are some proof it always converge to some stationary
+    point (i.e. a point $x^*$ for which $\nabla f(x^*) = 0$).}
+  \label{tab:minimize-unconstrained}
+\end{table}
+
+
+The \texttt{minimize} function provides methods for solving general non-linear optimization
+problems. It unifies several methods under the same general interface in order to facilitate
+interchanging between solvers with different characteristics.  Table~\ref{tab:minimize-unconstrained} summarizes
+all the methods available for solving unconstrained minimization problems of the type:
+\begin{equation}
+  \label{eq:unconstrained-minimization-prob}
+  \min_x f(x)
+\end{equation}
+where $f$ is a multivariate function $f: \mathbb{R}^n \rightarrow \mathbb{R}$ and $x \in \mathbb{R}^n$.
+
+There are a sufficient variety of methods to enable the user to solve problems with the diverse properties.
+Derivative-free methods (e.g. \texttt{Nelder-Mead} e \texttt{Powell}) are appropriate for minimizing noisy objective
+functions. When the derivatives are available, on the other hand, methods with faster convergence can also be used.
```

Comment:
"the derivatives" -> "clean derivatives"?  Here I'm using "clean" to indicate that I mean in the case when the function is not noisy. Or actually, it would probably be better if you wrote "are appropriate for minimizing non-differentiable or noisy functions objective functions. When derivatives are available..."
" methods with faster convergence can also be used." -> "other methods offer faster convergence."

Reply:
"e" --> "and". Also you might want to mention that derivative-free methods usually are heuristics without any guarantee on solution quality.

Reply:
I reformulated the phrase as: "Derivative-free methods (e.g. \texttt{Nelder-Mead} and \texttt{Powell}) are appropriate for minimizing non-differentiable or
noisy objective functions. When the derivatives are available, however, other methods offer faster convergence and have guarantees of convergence."

I think the fact that derivative-free methods do not have guarantees of convergence is implicit from the fact "other methods ... have guarantees of convergence"

Reply:
> "e" --> "and"

Sorry about that. "e" is the word for "and" in portuguese. I mixed things up.

Reply:
This is already much better. Conisder "other methods offer faster convergence and have guarantees of convergence" -> "other methods offer faster and guaranteed convergence."